packages/lib/src/spatial-navigation/components/virtualizedList/helpers/getNumberOfItemsVisibleOnScreen.ts

```diff
@@ -0,0 +1,43 @@
+const getMinSizeOfItems = <T>({
+  data,
+  itemSize,
+}: {
+  data: T[];
+  itemSize: number | ((item: T) => number);
+}) => {
+  if (typeof itemSize === 'number') {
+    return itemSize;
+  }
+
+  let minSize: number | undefined = undefined;
+
+  for (let i = 0; i < data.length; i++) {
+    const item = data[i];
+    if (item !== undefined) {
+      const currentSize = itemSize(item);
+      if (minSize === undefined || currentSize < minSize) {
+        minSize = currentSize;
+      }
+    }
+  }
+
+  return minSize;
+};
+
+export const getNumberOfItemsVisibleOnScreen = <T>({
+  data,
+  listSizeInPx,
+  itemSize,
+}: {
+  data: T[];
+  listSizeInPx: number;
+  itemSize: number | ((item: T) => number);
+}) => {
+  const itemSizeToComputeRanges = getMinSizeOfItems({ data, itemSize });
+
+  if (!itemSizeToComputeRanges) {
+    return 0;
```

Comment:
Maybe we should throw or log an error message here? (I'm guessing if 1 item is of size 0, we don't see any items) Or is it handled elsewhere?

Reply:
In this line in particular, we return 0 for the number of elements visible on screen, so none will be displayed. We do not need to throw as empty list should be handled without crashing I guess ? But I agree that we should handle itemSize that equal zero, as it will return Infinity, which would be... unconventionnal for sure haha

Reply:
I added a check for empty list :)
Lists with an item whose size is equal to 0 will be handled too by : 
```
  if (!itemSizeToComputeRanges) {
    return 0;
  }
```
And in this case no item will be displayed. I added warning in this case too.